apps/extension/src/ui/domains/Swap/hooks/useSwapTokensModal.ts

```diff
@@ -0,0 +1,21 @@
+import { useStateObservable } from "@react-rxjs/core"
+import { SetStateAction } from "react"
+
+import { setSwapTokensModalIsOpen, swapTokensModalIsOpen$ } from "../state/swapTokensModalIsOpen"
+
+export const useSwapTokensModal = () => ({
+  isOpen: useStateObservable(swapTokensModalIsOpen$),
+  open,
+  close,
+  toggle,
+  setIsOpen,
+})
+
+const open = () => setIsOpen(true)
+const close = () => setIsOpen(false)
+const toggle = () => setIsOpen((v) => !v)
+
+const setIsOpen = async (isOpen: SetStateAction<boolean>) =>
+  setSwapTokensModalIsOpen(
+    typeof isOpen === "function" ? isOpen(await swapTokensModalIsOpen$.getValue()) : isOpen,
+  )
```

Comment:
We should come up with a standard pattern for this.

i.e. when we want `setState(prev => /* return next state using prev */)` but for state held in rxjs.